### PR TITLE
Create users controller

### PR DESCRIPTION
**Pull request by zombre98**
IMAGE 2022-11-18 17:58:40

### CODE REVIEW

File: app/views/users/index.jbuilder [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+json.users @users, :id, :emil
```
Comment (ubrisson) [inline app/views/users/index.jbuilder:2]: Is this a type ?

File: config/application.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@
 module Interview
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 6.1
+    config.load_defaults 6.0
```
Comment (ubrisson) [inline config/application.rb:25]: Should not be in this pull request.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def index
+    @users = User.all
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:user)
```
Comment (ubrisson) [inline app/controllers/users_controller.rb:19]: params.require(:user).permit(:email, :password)

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  def create
+    @user = User.create(params)
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:5]: `user_params`

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def update
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:8]: missing routes

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  def create
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:4]: missing view

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (huguesbr) [inline app/controllers/users_controller.rb:8]: missing view

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:3]: rescue_from? already setup?

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (huguesbr) [inline app/controllers/users_controller.rb:5]: `User.create!`

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  def create
+    @user = User.create(params)
+  end
+
+  def update
+    @user = User.create(params)
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:9]: `User.create!`

Comment (huguesbr) [reply app/controllers/users_controller.rb:9]: `@user.errors[:field]`

File: config/application.rb (hunk 2 above)
Comment (huguesbr) [inline config/application.rb:25]: to remove?

File: config/routes.rb [hunk 9]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   resources :posts, only: :index
+  resources :users, only: [:index, :create]
```
Comment (huguesbr) [inline config/routes.rb:4]: missing route `update`

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (huguesbr) [inline app/controllers/users_controller.rb:19]: which fields do you want to allow?

File: app/views/users/index.jbuilder (hunk 1 above)
Comment (huguesbr) [inline app/views/users/index.jbuilder:2]: typo

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (huguesbr) [inline app/controllers/users_controller.rb:3]: typo

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:1]: WHERE ARE THE SPECS!?

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  def create
+    @user = User.create(params)
+  end
+
+  def update
+    @user = User.create(params)
+  end
+
+  def index
+    @users = User.all
```
Comment (huguesbr) [inline app/controllers/users_controller.rb:13]: `.includes`

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (JulienPoncelet) [inline app/controllers/users_controller.rb:3]: Faute d'humain !

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  def create
+    @user = User.create(params)
+  end
```
Comment (JulienPoncelet) [inline app/controllers/users_controller.rb:6]: Attention utilisation du create sans le ! je conseil plutôt de partir sur quelque chose du genre
```
@user = User.build(params)
if @user.save
  # 
else
  #
end
```

File: app/controllers/users_controller.rb [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+  def index
+    @users = User.all
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:user)
+  end
```
Comment (JulienPoncelet) [inline app/controllers/users_controller.rb:20]: missing `:email` dans le permit

File: config/application.rb (hunk 2 above)
Comment (JulienPoncelet) [inline config/application.rb:25]: Pourquoi ? 🗡️

**Pull request closed without merging**